test-harness/tests/conftest.py

```diff
@@ -95,32 +95,16 @@ def syslog_mock(log_catcher):
 
 
 @pytest.fixture(scope='session')
-def dns_mock(log_catcher, navstar_ips, resolvconf_fixup):
-    """Set-up DNS mocks, both for agent AR (port 53) and master AR (port 61053)"""
-    m_61053 = DNSMock(log_catcher, port=61053)
-    m_61053.start()
+def dns_server_mock():
+    s = MesosLeaderDNSServer(
+        server_address=('127.0.0.1', 61053),
+        leader_ip='127.0.0.2',
+        )
+    s.start()
 
-    m_53 = DNSMock(log_catcher, port=53)
-    m_53.start()
+    yield s
 
-    yield (m_53, m_61053)
-
-    m_61053.stop()
-    m_53.stop()
-
-
-@pytest.fixture(scope='session')
-def navstar_ips():
```

Comment:
> we have not written all the tests yet

I understand that the tests you are referring to here are supposed to cover basic behavior.

Just noting: in case of Admin Router, "full coverage" is undefined if we talk about real-world usage scenarios. "full coverage" in terms of code coverage is great, but can easily lead to a wrong sense of confidence. Concepts, formalism plus fuzzing/chaos/statistical testing with a high sampling density are required to obtain the degree of confidence that we strive for.